src/Kunstmaan/NodeBundle/Helper/Services/PageCreatorService.php

```diff
@@ -92,121 +124,169 @@ public function setContainer(ContainerInterface $container = null)
      * Automatically calls the ACL + sets the slugs to empty when the page is an Abstract node.
      *
      * @return Node The new node for the page.
-     *
-     * @throws \InvalidArgumentException
      */
-    public function createPage(HasNodeInterface $pageTypeInstance, array $translations, array $options = array())
+    public function createPage(HasNodeInterface $pageTypeInstance, array $translations, array $options = [])
     {
-        if (is_null($options)) {
-            $options = array();
-        }
+        Assert::notEmpty($translations, 'There has to be at least 1 translation in the translations array');
+        $this->pageTypeInstance = $pageTypeInstance;
+        $this->options = (null === $options) ? [] : $options;
+        $this->seo = $this->getSeo();
+        $this->creator = $this->getCreator();
+        $this->parent = $this->getOption('parent');
 
-        if (is_null($translations) || (count($translations) == 0)) {
-            throw new \InvalidArgumentException('There has to be at least 1 translation in the translations array');
+        $firstTranslation = array_shift($translations);
+        $this->handleTranslation($firstTranslation, true);
+
+        foreach ($translations as $translation) {
+            $this->handleTranslation($translation);
         }
 
-        $em = $this->entityManager;
+        $this->aclPermissionCreatorService->createPermission($this->rootNode);
+        return $this->rootNode;
+    }
 
-        /** @var NodeRepository $nodeRepo */
-        $nodeRepo = $em->getRepository('KunstmaanNodeBundle:Node');
-        /** @var $userRepo UserRepository */
-        $userRepo = $em->getRepository($this->userEntityClass);
+    /**
+     * @return UserRepository
+     */
+    private function getUserRepo()
+    {
+        /** @var UserRepository $userRepo */
+        $userRepo = $this->entityManager->getRepository($this->userEntityClass);
+        return $userRepo;
+    }
+
+    /**
+     * @return \Kunstmaan\SeoBundle\Entity\Seo|null
+     */
+    private function getSeo()
+    {
+        $seo = null;
         /** @var $seoRepo SeoRepository */
         try {
-            $seoRepo = $em->getRepository('KunstmaanSeoBundle:Seo');
-        } catch (ORMException $e) {
+            $seoRepo = $this->entityManager->getRepository('KunstmaanSeoBundle:Seo');
+            if (!is_null($seoRepo)) {
```

Comment:
If statement not necessary, catch `ORMException` instead like on original L115-119